src/components/filters.tsx

```diff
@@ -441,6 +474,17 @@ export const Filters = React.memo(function Filters({ torrents, currentFilters, s
                 >
                     Status filters
                 </Menu.Item>
+                <Menu.Divider/>
+                <Menu.Item
+                    icon={condensedDirTree ? <Icon.Check size="1rem" /> : <Box miw="1rem" />}
+                    onMouseEnter={closeStatusFiltersSubmenu}
+                    onMouseDown={(e) => {
+                        onCondensedDirTreeClick(!condensedDirTree);
+                        e.stopPropagation();
```

Comment:
This should be in the callback, that way you will not need a lambda here and can pass the callback directly as event handler.